pkg/sysfs/system.go

```diff
@@ -730,6 +750,16 @@ func (sys *system) discoverCPUs() error {
 		return err
 	}
 
+	// If we had partial core type overrides while discovering CPUs, some cores
+	// did not get their type set correctly. Fix that now.
+	for kind, ids := range sys.coreKindCPUs {
```

Comment:
Should we do this "hack" only if overrides were used. I.e. do this before the detection from sysfs and not tamper with what was detected from there?

Reply:
We can't do this before `checkCoreKinds()` is done as that might patch up the per kind cpusets for partial overrides. We could do this as part of `checkCoreKinds()` though, although it would be a bit more convoluted because of multiple successful code paths.

Reply:
> We can't do this before `checkCoreKinds()` is done as that might patch up the per kind cpusets for partial overrides.

K, got that

> We could do this as part of `checkCoreKinds()` though, although 

Yeah, it feels like the right place for this as it's doing other fixes as well

> it would be a bit more convoluted because of multiple successful code paths.

Are there really? There is this one `if len(sys.coreKindCPUs) == 0` but that branch shouldn't need any patching?

Reply:
Hmm, that is true. Lemme check if I can fold the fixup loop into `checkCoreKinds()`.

Reply:
Okay, I moved all verification and fixup into `checkCoreKinds()`.